packages/plugin-ext/src/plugin/authentication-ext.ts

```diff
@@ -0,0 +1,207 @@
+            } else {
+                await this.proxy.$requestNewSession(providerId, scopes, extensionId, extensionName);
+                return undefined;
+            }
+        }
+    }
+
+    async logout(providerId: string, sessionId: string): Promise<void> {
+        const provider = this.authenticationProviders.get(providerId);
+        if (!provider) {
+            return this.proxy.$logout(providerId, sessionId);
+        }
+
+        return provider.logout(sessionId);
+    }
+
+    registerAuthenticationProvider(provider: theia.AuthenticationProvider): theia.Disposable {
+        if (this.authenticationProviders.get(provider.id)) {
+            throw new Error(`An authentication provider with id '${provider.id}' is already registered.`);
+        }
+
+        this.authenticationProviders.set(provider.id, provider);
+        if (this._providerIds.indexOf(provider.id) === -1) {
+            this._providerIds.push(provider.id);
+        }
+
+        if (!this._providers.find(p => p.id === provider.id)) {
+            this._providers.push({
+                id: provider.id,
+                label: provider.label
+            });
+        }
+
+        const listener = provider.onDidChangeSessions(e => {
+            this.proxy.$updateSessions(provider.id, e);
+        });
+
+        this.proxy.$registerAuthenticationProvider(provider.id, provider.label, provider.supportsMultipleAccounts);
+
+        return new Disposable(() => {
+            listener.dispose();
+            this.authenticationProviders.delete(provider.id);
+            const index = this._providerIds.findIndex(id => id === provider.id);
+            if (index > -1) {
+                this._providerIds.splice(index);
+            }
+
+            const i = this._providers.findIndex(p => p.id === provider.id);
+            if (i > -1) {
+                this._providers.splice(i);
+            }
+
+            this.proxy.$unregisterAuthenticationProvider(provider.id);
+        });
+    }
+
+    $login(providerId: string, scopes: string[]): Promise<theia.AuthenticationSession> {
+        const authProvider = this.authenticationProviders.get(providerId);
+        if (authProvider) {
+            return Promise.resolve(authProvider.login(scopes));
+        }
+
+        throw new Error(`Unable to find authentication provider with handle: ${providerId}`);
+    }
+
+    $logout(providerId: string, sessionId: string): Promise<void> {
+        const authProvider = this.authenticationProviders.get(providerId);
+        if (authProvider) {
+            return Promise.resolve(authProvider.logout(sessionId));
+        }
+
+        throw new Error(`Unable to find authentication provider with handle: ${providerId}`);
+    }
+
+    $getSessions(providerId: string): Promise<ReadonlyArray<theia.AuthenticationSession>> {
+        const authProvider = this.authenticationProviders.get(providerId);
+        if (authProvider) {
+            return Promise.resolve(authProvider.getSessions());
```

Comment:
You can't send an object you got from the session provider over the wire: it might not serialize correctly. You need to make a copy into a well known structure. here.

Reply:
Done, added wrapping to a new object